tests/test_lora.py

```diff
@@ -261,6 +263,75 @@ def test_lora_layer_forward_no_exception(apply_to):
     model(input_ids)
 
 
+@torch.inference_mode()
+@pytest.mark.parametrize(
+    ("n_query_groups", "apply_to"),
+    list(
+        product(
+            (1, 2, 3, 6),
+            product((False, True), repeat=3),
+        ),
+    ),
+)
+def test_lora_gpt_query_groups_merge_and_forward_no_exception(n_query_groups, apply_to):
+    from lit_gpt.lora import GPT, Config, merge_lora_weights
+
+    keys = ("to_query", "to_key", "to_value")
+    values = apply_to
+    apply_to = dict(zip(keys, values))
+
+    config = Config(
+        n_layer=1,
+        n_head=6,
+        n_embd=12,
+        block_size=1,
+        vocab_size=1,
+        r=2,
+        alpha=8,
+        dropout=0.1,
+        n_query_groups=n_query_groups,
+        **apply_to,
+    )
+    model = GPT(config)
+    merge_lora_weights(model)
+    input_ids = torch.tensor([[1]])
+    model(input_ids)
+
+
+@torch.inference_mode()
+@pytest.mark.parametrize(
+    ("n_head", "enable_lora"),
+    list(
+        product(
+            (1, 2, 3, 6, 12),
+            [p for p in product((False, True), repeat=3) if any(p)] # skip enable_lora=(False, False, False)
+        ),
+    ),
+)
```

Comment:
```suggestion
@pytest.mark.parametrize("n_head", (1, 2, 3, 6, 12))
@pytest.mark.parametrize(
    "enable_lora",
    [
        (False, False, True),
        (False, True, False),
        (False, True, True),
        (True, False, False),
        (True, False, True),
        (True, True, False),
        (True, True, True),
    ],
)
```

Reply:
Didn't know that it's possible. 
Thanks!